test/dropout_util.hpp

```diff
@@ -244,32 +244,32 @@ inline void ExpandTensorDim(std::vector<T> x_len,
                             std::vector<T>& out_len,
                             std::vector<T>& out_str)
 {
-    auto itr_xl = x_len.end() - 1;
-    auto itr_yl = y_len.end() - 1;
-    auto itr_xs = x_str.end() - 1;
-    auto itr_ys = y_str.end() - 1;
-    auto itr_il = in_len.end() - 1;
-    auto itr_ol = out_len.end() - 1;
-    auto itr_is = in_str.end() - 1;
-    auto itr_os = out_str.end() - 1;
+    int xl_idx = x_len.size() - 1;
```

Comment:
Same here. Please ensure the Linux build does not trigger a warning in the first place.